source/connections-survive-step-down/tests/README.rst

```diff
@@ -0,0 +1,209 @@
+
+1. Connect to the server E and retrieve current replica set configuration using
+   the `replSetGetConfig <https://docs.mongodb.com/manual/reference/command/replSetGetConfig/>`_
+   admin command.
+2. For each item in the ``members`` field of the configuration, examine the
+   value of the ``host`` field. If the value matches the address of the
+   server N, set ``priority`` of the respective item to 10. If the value
+   matches the address of the server E, set ``priority`` to 1. Otherwise
+   set ``priority`` to 0.
+3. Increment the ``version`` field of the replica set configuration.
+4. Reconfigure the replica set using the `replSetReconfig
+   <https://docs.mongodb.com/manual/reference/command/replSetReconfig/>`_
+   command, providing the modified configuration.
+5. Directly connect to the server N and execute `replSetFreeze
+   <https://docs.mongodb.com/manual/reference/command/replSetFreeze/>`_
+   admin command as follows: ``{replSetFreeze: 0}``. If this command
+   fails with operation failure code 95 ("cannot freeze node when primary or
+   running for election. state: Primary"), perform server selection to
+   discover the current primary. If the current primary is server N, stop
+   as the procedure is complete. Otherwise propagate the operation failure
+   error(*).
+6. Connect to the server E and execute the `replSetStepDown
+   <https://docs.mongodb.com/manual/reference/command/replSetStepDown/>`_
+   admin command as follows:
+   ``{replSetStepDown: 4, secondaryCatchUpPeriodSecs: 2}``.
+7. Connect directly to the server N and execute the following admin command:
+   ``{replSetStepUp: 1}``. If this command fails with an operation failure
+   error with code 125 ("Election failed"), repeat this step 7.
+8. Trigger a rescan of the topology, such as by setting the status of all
+   servers in the topology to Unknown.
+9. Perform server selection to obtain the current primary.
+10. If the current primary is not N, go to step 7 and step up the server N
+    again.
+
+(*) The cluster may have held an election after the replica set was reconfigured
+in step 4, making the server N the current primary. If this happens,
+``replSetFreeze`` command will fail, but the overall goal of having the
+server N as the primary has been achieved.
+
+As the above procedure contains a potentially infinite loop, the driver MAY
+impose a time limit and fail if the server N has not become a primary in the
+allotted time. This time limit SHOULD be at least 10 seconds.
+
+Stepping Down A Server
+``````````````````````
+
+If the driver does not care which server becomes the new primary, as would be
+the case in the getMore test for example, the driver should perform the
+following procedure:
+
+- Obtain a list of servers in the cluster.
+- Randomly or otherwise choose a server to be the new primary, other than the
+  current primary.
+- Follow the above server election procedure to elect the chosen server as
+  the new primary.
+
+
+Questions and Answers
```

Comment:
For drivers that do not have pools and do not implement CMAP, I think it'd make sense to discuss an alternative method of asserting connection behavior here. IIRC, we talked about checking [`connections.totalCreated`](https://docs.mongodb.com/manual/reference/command/serverStatus/#serverstatus.connections.totalCreated) from `serverStatus` in a previous meeting.

Reply:
Added as a Q&A section, thank you.